### PR TITLE
Safety alert case 1152

### PR DESCRIPTION
**Pull request by amanbop**
API integration with Safety Alert Card

### CODE REVIEW

**Comment by SashaKyz** [comment]
Can one of the admins verify this patch?

File: app/javascript/Clients/clientIdPage/SafetyAlertInformation.js [hunk 1]
```diff
@@ -9,20 +11,71 @@ import {
...
-  { value: 'Yuba County', label: 'Yuba County' },
-  { value: 'Merced County', label: 'Merced County' },
-];
-const deactivationinCounty = [
-  { value: 'Los Angeles County', label: 'Los Angeles County' },
-  { value: 'Sacramento County', label: 'Sacramento County' },
-  { value: 'El Dorado County', label: 'El Dorado County' },
-  { value: 'Yuba County', label: 'Yuba County' },
-  { value: 'Merced County', label: 'Merced County' },
+const countyList = [
```
[the remark this answers is not in the document]
Comment (amanbop) [reply app/javascript/Clients/clientIdPage/SafetyAlertInformation.js]: I am going to put a story for this in the coming sprint. This is just a stop gap arrangement.

File: config/routes.rb [hunk 2]
```diff
@@ -12,8 +12,12 @@
     end
 
     resources :clients, only: [ :show ] do
+      member do
```
[the remark this answers is not in the document]
Comment (RamuKammagani-osi) [reply config/routes.rb]: I wonder how i missed it, Anyway its Done! now

File: spec/lib/childclients/child_clients_repository_spec.rb [hunk 3]
```diff
@@ -1,38 +0,0 @@
-# frozen_string_literal: true
-
```
[the remark this answers is not in the document]
Comment (RamuKammagani-osi) [reply spec/lib/childclients/child_clients_repository_spec.rb]: That was deleted as a part of deleting the duplicate files! I will add those back!

File: lib/clients/safety_alert.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    attribute :activation_date, Types::String.optional
+    attribute :activation_explanation_text_id, Types::String.optional
+    attribute :client_id, Types::String.optional
+    attribute :activation_reason_code, Types::String.optional
+    attribute :third_id, Types::String.optional
+    attribute :deactivation_government_entity_code, Types::String.optional
+    attribute :deactivation_date, Types::String.optional
+    attribute :deactivation_explanation_text_id, Types::String.optional
+  end
+end
```
[the remark this answers is not in the document]
Comment (RamuKammagani-osi) [reply lib/clients/safety_alert.rb]: Yup! I changed it back the singular

File: app/javascript/_services/client/client.service.js [hunk 5]
```diff
@@ -6,6 +6,11 @@ class ClientService {
       response => response.data
     );
   }
+  static fetchSafetyAlerts() {
```
[the remark this answers is not in the document]
Comment (edgibbs) [reply app/javascript/_services/client/client.service.js]: Leaving as static for now, may change with others in a future story.

File: spec/controllers/api/clients_controller_spec.rb [hunk 6]
```diff
@@ -25,6 +25,24 @@ module Api
         get :show, params: { id: 5 }
         expect(response.body).to eq client.to_json
       end
+
+      it 'has a route' do
```
Comment (edgibbs) [inline spec/controllers/api/clients_controller_spec.rb]: Just wrap in a describe '#safety_alerts'

File: app/controllers/api/clients_controller.rb [hunk 7]
```diff
@@ -3,8 +3,14 @@
...
   class ClientsController < ActionController::API
     def show
-      client = Clients::ClientRepository.new.show(params[:id], session[:token])
-      render json: client
+      clients = Clients::ClientRepository.new.show(params[:id], session[:token])
+      render json: clients
+    end
+
+    def safety_alerts
+      client_safety_alert_repo = Clients::ClientRepository.new
```
[the remark this answers is not in the document]
Comment (edgibbs) [reply app/controllers/api/clients_controller.rb]: cool it does now

**Pull request merged**